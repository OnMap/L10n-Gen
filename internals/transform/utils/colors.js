// More here -> https://stackoverflow.com/questions/9781218/how-to-change-node-jss-console-font-color
const COLORS = {
  UNDERSCORE: '\x1b[4m',

  RED: '\x1b[31m',
  GREEN: '\x1b[32m',
  CYAN: '\x1b[36m',
  YELLOW: '\x1b[33m',
  MAGENTA: '\x1b[35m',
};

Object.keys(COLORS).forEach((key) => {
  COLORS[key] = `${COLORS[key]}%s\x1b[0m`;
});

module.exports = COLORS;
