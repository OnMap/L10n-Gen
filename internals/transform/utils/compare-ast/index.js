// Taken from https://github.com/jugglinmike/compare-ast

const parseModule = require('esprima').parseModule;
const Errors = require('./match-error');
const normalizeMemberExpr = require('./util/normalize-member-expr');
const fuzzyIdentifiers = require('./comparators/fuzzy-identifiers');
const fuzzyStrings = require('./comparators/fuzzy-strings');

const checkEquivalency = function (comparators, actual, expected) {
  const result = comparators
    .map(comparator => comparator(actual, expected))
    .reduce((prev, current) => {
      if (current === true) {
        return true;
      }
      return prev || current;
    }, null);

  if (result instanceof Errors) {
    throw result;
  } else if (result === true) {
    return true;
  }
};

function compareAst(actualSrc, expectedSrc, options) {
  let actualAst;
  let expectedAst;
  options = options || {};

  if (!options.comparators) {
    options.comparators = [];
  }

  Array.prototype.push.apply(options.comparators, [
    fuzzyIdentifiers(options),
    fuzzyStrings(options)
  ]);

  try {
    actualAst = parseModule(actualSrc).body;
  } catch (err) {
    throw new Errors.ParseError();
  }

  try {
    expectedAst = parseModule(expectedSrc).body;
  } catch (err) {
    throw new Errors.ParseError();
  }

  function _bind(actual, expected) {
    let attr;

    // Literal values
    if (Object(actual) !== actual) {
      if (actual !== expected) {
        throw new Errors.MatchError(actualAst, expectedAst);
      }
      return;
    }

    // Arrays
    if (Array.isArray(actual)) {
      if (actual.length !== expected.length) {
        throw new Errors.MatchError(actualAst, expectedAst);
      }
      actual.forEach((_, i) => {
        _bind(actual[i], expected[i]);
      });
      return;
    }

    // Nodes

    normalizeMemberExpr(actual);
    normalizeMemberExpr(expected);

    if (checkEquivalency(options.comparators, actual, expected)) {
      return;
    }

    // Either remove attributes or recurse on their values
    for (attr in actual) {
      if (expected && attr in expected) {
        _bind(actual[attr], expected[attr]);
      } else {
        throw new Errors.MatchError(actualAst, expectedAst);
      }
    }
  }

  // Start recursing on the ASTs from the top level.
  _bind(actualAst, expectedAst);

  return null;
}

compareAst.Error = Errors;

module.exports = (source1, source2, options) => {
  try {
    compareAst(source1, source2, options);
  } catch (err) {
    return err;
  }
  return null;
};
