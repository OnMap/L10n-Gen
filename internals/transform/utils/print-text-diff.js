/*
  Prints the diff of two text files line by line (like git)
  More advanced web version - http://incaseofstairs.com/jsdiff/
*/

const diff = require('jest-diff');
const COLORS = require('./colors');

const printDiff = (was, now, { printNotChanged = true, printLegend = false, header = '' } = {}) => {
  const textDiff = diff(was, now);
  let printCalled = false;

  if (header) {
    console.log(COLORS.CYAN, header);
  }

  textDiff.split('\n').forEach((line) => {
    const first = line[0];
    if (first === '+' || first === '-') {
      const color = first === '+' ? COLORS.GREEN : COLORS.RED;
      if (!printLegend) {
        if (line === '- Expected' || line === '+ Received') {
          return;
        }
      }
      console.log(color, line);
      printCalled = true;
    } else if (printNotChanged) {
      console.log(line);
      printCalled = true;
    }
  });

  if (!printCalled) {
    console.log(COLORS.CYAN, 'Files are identical. No diff.');
  }
};

module.exports = printDiff;
