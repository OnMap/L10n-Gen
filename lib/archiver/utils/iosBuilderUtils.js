const toPairs = require('lodash/toPairs');

const joinPairToLine = (pair = []) => pair
  .map(item => `"${item}"`)
  .join(' = ');

const reduceTranslationsToStringForIOS = (translationsObject = {}) => {
  const arrayOfPairs = toPairs(translationsObject);

  return arrayOfPairs
    .map(joinPairToLine)
    .join(';\n') + ';';
};

module.exports = {
  joinPairToLine,
  reduceTranslationsToStringForIOS
};
