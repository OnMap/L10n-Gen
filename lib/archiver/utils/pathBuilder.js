const {
  translationsFileTypes,
  translationsFileNames,
  translationsDirectoryNames
} = require('../../constants/index');

const buildFilePath = (appType, language) => {
  let directory = '';
  const fileName = translationsFileNames[appType];
  const dirName = translationsDirectoryNames[appType];


  switch (appType) {
    case translationsFileTypes.IOS: {
      directory = `${language}${dirName}`;
      break;
    }
    default: {
      directory = language;
    }
  }
  return `${directory}/${fileName}`;
};

module.exports = buildFilePath;
