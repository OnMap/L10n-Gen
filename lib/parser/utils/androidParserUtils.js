const get = require('lodash/get');
const { xmlParserOptions, xmlBuilderOptions } = require('../../constants/index');

const reduceAndroidTranslationsPairs = (array = []) => array
  .reduce((result, item) => {
    const key = get(item, `${xmlParserOptions.attrkey}.${xmlBuilderOptions.attributeName}`);
    const value = get(item, xmlParserOptions.charkey) || '';

    if (key) {
      result.push([ key, value ]);
    }

    return result;
  }, []);

module.exports = {
  reduceAndroidTranslationsPairs
};
