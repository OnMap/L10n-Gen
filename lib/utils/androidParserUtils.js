const get = require('lodash/get');
const { xmlParserOptions } = require('../constants/index');

const reduceAndroidTranslationsPairs = (array = []) => array
  .reduce((result, item) => {
    const key = get(item, `${xmlParserOptions.attrkey}.name`);
    const value = get(item, xmlParserOptions.charkey) || '';

    if (key) {
      result.push([ key, value ]);
    }

    return result;
  }, []);

module.exports = {
  reduceAndroidTranslationsPairs
};
