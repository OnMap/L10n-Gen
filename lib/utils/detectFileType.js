const { translationsFileNames } = require('../constants');

const detectTranslationsFileType = (path = '') => {
  const fileName = path.split('/').pop();
  const detected = translationsFileNames.find(item => (item.fileName === fileName));
  return (detected) ? detected.type : null;
};

module.exports = detectTranslationsFileType;
