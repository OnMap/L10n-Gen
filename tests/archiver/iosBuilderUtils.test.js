import { joinPairToLine, reduceTranslationsToStringForIOS } from '../../src/archiver/utils/iosBuilderUtils';

const translationsPair = ['Account.CancelBarButton', 'Cancel'];
const translationsObject = {
  'Account.CancelBarButton': 'Cancel',
  'Account.Login.EmailErrorLabel': 'Enter a valid email address',
  'Account.Login.EmailTextField': 'Email address',
  'Account.Login.ForgotButton': 'Forgot password?',
  'Account.Login.LoginButton': 'Login'
};

describe('Builder utils for iOS', () => {
  it('reduce pair array to string for iOS', () => {
    expect(joinPairToLine(translationsPair)).toMatchSnapshot();
  });

  it('reduce iOS translations from object', () => {
    expect(reduceTranslationsToStringForIOS(translationsObject)).toMatchSnapshot();
  });
});

