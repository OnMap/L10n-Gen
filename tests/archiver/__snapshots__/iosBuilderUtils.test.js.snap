// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Builder utils for iOS reduce iOS translations from object 1`] = `
"\\"Account.CancelBarButton\\" = \\"Cancel\\";
\\"Account.Login.EmailErrorLabel\\" = \\"Enter a valid email address\\";
\\"Account.Login.EmailTextField\\" = \\"Email address\\";
\\"Account.Login.ForgotButton\\" = \\"Forgot password?\\";
\\"Account.Login.LoginButton\\" = \\"Login\\";"
`;

exports[`Builder utils for iOS reduce pair array to string for iOS 1`] = `"\\"Account.CancelBarButton\\" = \\"Cancel\\""`;
